### PR TITLE
delete BucketClass

### PR DESCRIPTION
**Pull request by naveenpaul1**
- when BucketClass deletes respective OBC will get deleted automatically
- request and limit are added to kube proxy
 [DFMS-152](https://issues.redhat.com/browse/DFMS-152)

Sign off by Naveen Paul @naveenpaul1

### CODE REVIEW

**Comment by rexagod** [comment]
/override ci/prow/lint
Also opened an issue to track this here: https://github.com/golangci/golangci-lint/issues/2960

File: utils/resources.go [hunk 1]
```diff
@@ -62,4 +62,14 @@ var ResourceRequirements = map[string]corev1.ResourceRequirements{
...
+	"kube-rbac-proxy": {
+		Limits: corev1.ResourceList{
+			"memory": resource.MustParse("30Mi"),
+			"cpu":    resource.MustParse("50m"),
+		},
+		Requests: corev1.ResourceList{
+			"memory": resource.MustParse("30Mi"),
+			"cpu":    resource.MustParse("50m"),
+		},
+	},
```
Comment (rexagod) [inline utils/resources.go:74]:
```yaml
        resources:
          limits:
            cpu: 500m
            memory: 128Mi
          requests:
            cpu: 5m
            memory: 64Mi
```
This is the current resource requirement specified by OSDK in `config/` for `kube-rbac-proxy` right now. Are we sure that our KRP usage will not exceed `30Mi` and `50m`?

Comment (naveenpaul1) [reply utils/resources.go:74]: @rexagod  this is the configuration that we have in ocs-osd deployer, https://github.com/red-hat-storage/ocs-osd-deployer/blob/main/utils/resources.go#L211
I personally think limits in OSDK in config/are very high for a sidecar

Comment (rexagod) [reply utils/resources.go:74]: ACK!

**Pull request merged**